python_modules/libraries/dagster-components/dagster_components/core/component.py

```diff
@@ -283,50 +288,51 @@ def current() -> "ComponentLoadContext":
     def for_test(
         *,
         resources: Optional[Mapping[str, object]] = None,
-        decl_node: Optional[ComponentDeclNode] = None,
+        decl_node: Optional["DefsModuleDecl"] = None,
     ) -> "ComponentLoadContext":
         return ComponentLoadContext(
-            module_name="test",
+            defs_root=Path.cwd(),
+            defs_module_name="test",
             decl_node=decl_node,
-            resolution_context=ResolutionContext.default(
-                decl_node.get_source_position_tree() if decl_node else None
-            ),
+            resolution_context=ResolutionContext.default(),
             module_cache=DefinitionsModuleCache(resources=resources or {}),
         )
 
     @property
     def path(self) -> Path:
-        from dagster_components.core.component_decl_builder import (
-            ComponentFolder,
-            ImplicitDefinitionsComponentDecl,
-            PythonComponentDecl,
-            YamlComponentDecl,
-        )
-
-        if not isinstance(
-            self.decl_node,
-            (
-                YamlComponentDecl,
-                PythonComponentDecl,
-                ImplicitDefinitionsComponentDecl,
-                ComponentFolder,
-            ),
-        ):
-            check.failed(f"Unsupported decl_node type {type(self.decl_node)}")
-
-        return self.decl_node.path
+        return check.not_none(self.decl_node).path
 
     def with_rendering_scope(self, rendering_scope: Mapping[str, Any]) -> "ComponentLoadContext":
         return dataclasses.replace(
             self,
             resolution_context=self.resolution_context.with_scope(**rendering_scope),
         )
 
-    def for_decl_node(self, decl_node: ComponentDeclNode) -> "ComponentLoadContext":
-        return dataclasses.replace(self, decl_node=decl_node)
+    def for_decl(self, decl: "DefsModuleDecl") -> "ComponentLoadContext":
+        return dataclasses.replace(self, decl_node=decl)
+
+    def defs_relative_module_name(self, path: Path) -> str:
+        """Returns the name of the python module at the given path, relative to the project root."""
+        container_path = self.path.parent if self.path.is_file() else self.path
```

Comment:
The previous implementation of this function was also the source of some annoying bugs. It was written in such a way that worked with the incorrect paths that were provided to it by the old system, so once I fixed the fact that we were passing the wrong paths down to the load context, the old logic started breaking.

I think this setup is a lot cleaner than the previous one.